src/shogun/features/RandomKitchenSinksDotFeatures.cpp

```diff
@@ -37,14 +38,13 @@ SGMatrix<float64_t> RandomKitchenSinksDotFeatures::generate_random_coefficients(
 {
 	SGVector<float64_t> vec = generate_random_parameter_vector();
 	SGMatrix<float64_t> random_params(vec.vlen, num_samples);
-	for (index_t dim=0; dim<random_params.num_rows; dim++)
-		random_params(dim, 0) = vec[dim];
+	linalg::range_fill(random_params,0.0);
+	linalg::add_col_vec(random_params,0,vec,random_params);
```

Comment:
I was wondering, why did we separate the case for col(0) when it can be put in the loop below?